src/js/components/BreadcrumbSegment.js

```diff
@@ -0,0 +1,54 @@
+import React, {PropTypes} from 'react';
+
+import BreadcrumbSegmentLink from './BreadcrumbSegmentLink';
+
+class BreadcrumbSegment extends React.Component {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+      isLoadingCrumb: true
+    };
+
+    this.store_listeners = [];
+  }
+
+  getCrumbLabel() {
+    return '';
+  }
+
+  getBackupCrumbLabel() {
```

Comment:
I have a hard time reasoning about what is going on here in this function. Could you add an explanatory comment in the code?